multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -0,0 +1,232 @@
+		return err
+	}
+
+	resImpList := &mcsv1alpha1.ResourceImportList{}
+	if err := remoteCluster.List(ctx, resImpList, &client.ListOptions{Namespace: remoteCluster.GetNamespace()}); err != nil {
+		return err
+	}
+
+	svcImpItems := svcImpList.Items
+	var mcsSvcItems []corev1.Service
+	for _, svc := range svcList.Items {
+		if _, ok := svc.Annotations[common.AntreaMCServiceAnnotation]; ok {
+			mcsSvcItems = append(mcsSvcItems, svc)
+		}
+	}
+
+	for _, resImp := range resImpList.Items {
+		for k, svc := range mcsSvcItems {
+			if svc.Name == common.AntreaMCSPrefix+resImp.Spec.Name && svc.Namespace == resImp.Spec.Namespace {
+				// Set the valid Service item as empty Service, then all left non-empty items should be removed.
+				mcsSvcItems[k] = corev1.Service{}
+			}
+		}
+
+		for n, svcImp := range svcImpItems {
+			if svcImp.Name == resImp.Spec.Name && svcImp.Namespace == resImp.Spec.Namespace {
+				svcImpItems[n] = k8smcsv1alpha1.ServiceImport{}
+			}
+		}
+	}
+
+	for _, svc := range mcsSvcItems {
+		s := svc
+		if s.Name != "" {
+			klog.InfoS("clean up Service", "service", klog.KObj(&s))
+			if err := c.Client.Delete(ctx, &s, &client.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+	}
+
+	for _, svcImp := range svcImpItems {
+		si := svcImp
+		if si.Name != "" {
+			klog.InfoS("clean up ServiceImport", "serviceimport", klog.KObj(&si))
+			if err = c.Client.Delete(ctx, &si, &client.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+	}
+
+	// clean up any ResourceExport if no corresponding ServiceExport locally
+	resExpList := &mcsv1alpha1.ResourceExportList{}
+	if err := remoteCluster.List(ctx, resExpList, &client.ListOptions{Namespace: remoteCluster.GetNamespace()}); err != nil {
+		return err
+	}
+	svcExpList := &k8smcsv1alpha1.ServiceExportList{}
+	if err := c.List(ctx, svcExpList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	resExpItems := resExpList.Items
+	svcExpItems := svcExpList.Items
+
+	for k, re := range resExpItems {
+		for _, se := range svcExpItems {
+			if re.Spec.Name == se.Name && re.Spec.Namespace == se.Namespace {
+				// Set the valid ResourceExport item as empty ResourceExport, then all left non-empty items should be removed.
+				resExpItems[k] = mcsv1alpha1.ResourceExport{}
+			}
+		}
+	}
+
+	for _, r := range resExpItems {
+		re := r
+		if re.Name != "" {
+			klog.InfoS("clean up ResourceExport", "ResourceExport", klog.KObj(&re))
+			if err := remoteCluster.Delete(ctx, &re, &client.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Enqueue will be called after StaleController is initialized.
+func (c *StaleController) Enqueue() {
+	// The key can be anything as we only have single item.
+	c.queue.Add("key")
+}
+
+// Run starts the StaleController and blocks until stopCh is closed.
+func (c *StaleController) Run(stopCh <-chan struct{}) {
```

Comment:
Add a comment saying we just run one to clean up stale resources.

Reply:
done

Reply:
done